python/ray/serve/_private/deployment_scheduler.py

```diff
@@ -462,5 +585,95 @@ def key(node_and_num_running_replicas_of_all_deployments):
 
         return replicas_to_stop
 
+    def _schedule_replica_compactly(
+        self,
+        scheduling_request: ReplicaSchedulingRequest,
+    ):
+        replica_name = scheduling_request.replica_name
+        deployment_id = scheduling_request.deployment_id
+        if (
+            scheduling_request.placement_group_bundles
+            and scheduling_request.placement_group_strategy == "STRICT_PACK"
+        ):
+            bundle_resources = [
+                Resources(bundle)
+                for bundle in scheduling_request.placement_group_bundles
+            ]
+            required_resources = sum(bundle_resources, Resources())
+        else:
+            required_resources = Resources(scheduling_request.actor_resources)
```

Comment:
this looks redundant with the sorting logic above -- possible to share the code for it?

Reply:
Moved to shared property of `ReplicaSchedulingRequest`.